File: Data Structure/Lab 4/Delete.c
//Write a program to delete a number at a given location in an array. 

#include<stdio.h>
void main(){
    int n,a[n],i = 0,index,val;

    printf("Enter size of an aay(n) :");
    scanf("%d",&n);

    printf("Enter index and value :");
    scanf("%d",&index);
    scanf("%d",&val);

    for(i = 0; i < n ; i++){
        printf("Enter element :");
        scanf("%d",&a[i]);
    }

    if(index >= n || index < 0){
        printf("Invalid Index !!");
    }
    else{
        //Shifting an element(Left)
        for(i = index ; i <= n - 1; i++){
            a[index] = a[index + 1];
        }
    }

    n--;

    for(i = 0; i < n ; i++){
        printf("%d\n,",a[i]);
    }
}

File: Data Structure(Java)/Lab 3/Delete.java
// Write a program to delete a number from an array that is already sorted in an 
//ascending order. 

import java.util.Scanner;

public class Delete {
    public static void main(String[] args) {
         
        Scanner sc = new Scanner(System.in);

        System.out.print("Enter size of an array :");
        int n = sc.nextInt();
        int[] arr = new int[n];

        for(int i = 0 ; i < n ; i++){
            System.out.println("Enter array element at "+i);
            arr[i] = sc.nextInt();
        }

        System.out.println("Final array :");
        for(int i = 0; i < n ; i++){
            if(i == n - 1){
                System.out.print(arr[i]);
            }
            else{
                System.out.print(arr[i]+",");
            }
        }

        System.out.print("Enter element you want to delete :");
        int ele = sc.nextInt();

        int index = n;

        for(int i = 0 ; i < n;  i++){
            if(arr[i] == ele){
                index = i;
                break;
            }
        }

        if(index < n){
            for(int i = index; i < n - 1; i++){
                arr[i] = arr[i + 1];
            }

            n--;

            for(int i = 0; i < n ; i++){
                if(i == n - 1){
                    System.out.print(arr[i]);
                }
                else{
                    System.out.print(arr[i]+",");
                }
            }

        }
        else{
            System.out.println("Element not found!!");
        }

        sc.close();
    }
}
